### DISCUSSION TITLE
Calling gl.users.list with a username for someone who doesn't exist

### DISCUSSION THREAD

**Discussion by willnwalker** [python-gitlab/python-gitlab#2232] [Q&A]
Hi all,

When attempting to run our code fragment which tries to add a user to a newly-created repository, we get an exception whenever that user does not yet exist (as expected). However, when the user goes to sign in via SSO and an account is created for them for the first time, their username has a 1 appended to it (seemingly due to a conflict from when the script searched for the user on our Gitlab instance). Has anyone else experienced this weird behavior before? We are trying to determine if the fault lies with our script, python-gitlab, or Gitlab. I was unable to find anything in the issues for this project or Gitlab.

`        try:
            u = gl.users.list (username=user)[0]
            allowed_users.append (u.id)`
`        except:
            print ('    {0}: User {1} doesn\'t (yet) exist ... skipping'.format (repo_fullname, user))`

**Comment by nejch** [comment]
@willnwalker could you format your code and maybe add a little more for context?

Generally project membership does not have much to do with the actual user creation at instance-level so I don't see that being an issue with python-gitlab unless you also use it to create users at instance level. I'd say it's likely more to do with how external identities are used to create the user with SSO. A little more code would help us get a better idea though!

**Reply by willnwalker** [reply]
Here's the rest of the source code, for context. Instructors at our university use this to create repositories for students to store their assignment submissions.

**Reply by nejch** [reply]
@willnwalker I wish I could help you here but I don't think python-gitlab is at fault as it's only calling GET endpoints for users which shouldn't really have an effect, or if it does, it would be an upstream issue. you can of course add `gl.enable_debug()` after line 205 to see more output on what's sent to the server if it helps your ops team.

You could try to reproduce a minimal example by calling the offending endpoint with a non-existent user which seems to be `GET /users?username=nonexistent` (https://docs.gitlab.com/ee/api/users.html#list-users) to see if that has a side effect on your auth provider but I'm not sure how that would be the case - I think you might need to look at that integration though.

**Reply by willnwalker** [reply]
Thank you for the feedback! I will try sending a request to that endpoint manually and see if it reproduces the behavior we're seeing.
